src/commands/secret/mod.rs

```diff
@@ -0,0 +1,141 @@
+use cloudflare::endpoints::workers::{CreateSecret, CreateSecretParams, DeleteSecret, ListSecrets};
+use cloudflare::framework::apiclient::ApiClient;
+use cloudflare::framework::response::ApiFailure;
+use cloudflare::framework::HttpApiClientConfig;
+
+use crate::http;
+use crate::settings::global_user::GlobalUser;
+use crate::settings::toml::Target;
+use crate::terminal::{emoji, interactive, message};
+
+fn format_error(e: ApiFailure) -> String {
+    http::format_error(e, Some(&secret_errors))
+}
+
+fn validate_target(target: &Target) -> Result<(), failure::Error> {
+    let mut missing_fields = Vec::new();
+
+    if target.account_id.is_empty() {
+        missing_fields.push("account_id")
+    };
+
+    if !missing_fields.is_empty() {
+        failure::bail!(
+            "{} Your wrangler.toml is missing the following field(s): {:?}",
+            emoji::WARN,
+            missing_fields
+        )
+    } else {
+        Ok(())
+    }
+}
+
+// secret_errors() provides more detailed explanations of Workers KV API error codes.
+// See https://api.cloudflare.com/#workers-secrets ? for details.
+fn secret_errors(error_code: u16) -> &'static str {
+    match error_code {
+        7003 | 7000 => {
+            "Your wrangler.toml is likely missing the field \"account_id\", which is required to write to Workers KV."
+        }
+        10001 => "The content you passed in is not an accepted string. Try entering just a string",
+        10053 => "There is already another binding with a different type by this name. Check your wrangler.toml or dashboard for conflicting bindings",
```

Comment:
so this happens in the  case that there is a binding by the same name but of another kind, so perhaps a kv namespace or plain text binding, for example. this isn't in the case of a secret by the same name, because this is a set so it'll just overwrite.

Reply:
ah. how about `"There is already a KV namespace or text binding with this name associated with your script"`or something like that/ "binding with a different type by this name" is hard to parse

Reply:
there is a world in which it is actually a text blob binding you are using for sites, or you went and called it `wasm` or something else that's taken by our build process. if we wanted to be really robust, we could query the list of bindings to show what's up, check their toml, look to see if they're stomping  on their own asset manifest or wasm binding... this feels like a good add-on issue though.

Reply:
👍